ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/how-to/custom-templates.md

```diff
@@ -0,0 +1,58 @@
+<!--
+  ~ Copyright (c) 2023 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Custom Templates
+
+With Custom Templates for the `eos_cli_config_gen` role, the user can create additional CLI configuration for features that are not covered by the built-in templates.
+
+## Create a Custom Template
+
+Custom templates should be written in Jinja2.
+
+For details on how to create Jinja2 templates, see [Templating Jinja2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_templating.html#templating-jinja2)
+
+Custom templates can utilize variables defined by all the AVD variable definition mechanisms. For more details see the explanation in [Custom Structured Configuration](../../../eos_designs/docs/how-to/custom-structured-configuration.md)
+
+If the required variable is not available by any of these mechanisms, it can be defined using a custom template in the `eos_designs` role. For more details see [Custom Templates for eos_designs](../../../eos_designs/docs/how-to/custom-templates.md)
```

Comment:
```suggestion
If the required variable is not available by any of these mechanisms, it can be defined using a custom template in the `eos_designs` role. For more details, see [Custom Templates for eos_designs](../../../eos_designs/docs/how-to/custom-templates.md)
```